docs/machine-learning/tutorials/movie-recommmendation.md

```diff
@@ -0,0 +1,479 @@
+
+While this is a good start, in reality you might want to add other attributes or `Features` (e.g. age, gender, geo-location, etc.) if they are included in the dataset. Adding more relevant `Features` can help improve the performance of your recommendation model. 
+
+If you are unsure about which `Features` might be the most relevant for your machine learning task, you can also make use of Feature Contribution Calculation (FCC) and [Feature Permutation Importance](../how-to-guides/determine-global-feature-importance-in-model.md), which ML.NET provides to discover the most influential `Features`.
+
+### Algorithm
+
+While ML.NET provides good default training algorithms, you can further fine-tune performance by changing parameters on these individual algorithms.
+
+For `Matrix Factorization`, you can experiment with Learning Rate, ApproximationRank, NumberOfIterations, and several other options to see if that gives you better results.
+
+Here is an example of how to fine-tune the `Matrix Factorization Trainer` options:
+
+```CSharp
+var options = new MatrixFactorizationTrainer.Options
+{
+  MatrixColumnIndexColumnName = "userIdEncoded",
+  MatrixRowIndexColumnName = "movieIdEncoded",
+  LabelColumnName = "Label",
+  NumberOfThreads = 1,
+  ApproximationRank = 32,
+  LearningRate = 0.3
+};
+```
+
+### New user scenario
+One common problem in collaborative filtering is the cold start problem, which is when you have a new user with no previous data to draw inferences from. This is often solved by asking new users to create a profile and, for instance, rate movies they have seen in the past. While this puts some burden on the user, it provides some starting data for new users with no rating history.
+
+## Other Recommendation Engines
+This is only one approach for performing movie recommendations. In many cases, you may not have the ratings data available and only have movie history available from users. In other cases, you may have more than just the user’s rating data.
+
+| Algorithm       | Scenario           | Sample  |
+| ------------- |:-------------:| -----:|
+| One Class Matrix Factorization | Use this when you only have userId and movieId. This style of recommendation is based upon the co-purchase scenario, or products frequently bought together, which means it will recommend to customers a set of products based upon their own purchase order history. | [>Try it out](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/MatrixFactorization_ProductRecommendation) |
+| Field Aware Factorization Machines | Use this to make recommendations when you have more Features beyond userId, productId, and rating (such as product description or product price). This also uses a collaborative filtering approach. | [>Try it out](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/end-to-end-apps/Recommendation-MovieRecommender) |
+
+## Other ML.NET Scenarios
```

Comment:
I have a question on this front, normally a targetted approach where a user progressively learns works well. In this particular case however the different ML tasks are all top of the funnel entry points which don't progressively build on top of each other. The risk we run with suggesting just one ML scenario/task is that it might not be interesting to a user. Is there a way to present a choice here and ask users to self select?

Reply:
This is a requirement of our content model.